packages/schema/src/environment/index.ts

```diff
@@ -6,14 +6,14 @@ export const EnvironmentSchema = z.object({
   name: z.string(),
   slug: z.string(),
   description: z.string().nullable(),
-  createdAt: z.string(),
-  updatedAt: z.string(),
+  updatedAt: z.string().datetime(),
+  createdAt: z.string().datetime(),
   lastUpdatedById: z.string(),
   projectId: z.string()
 })
 
 export const CreateEnvironmentRequestSchema = z.object({
-  name: z.string(),
+  name: EnvironmentSchema.shape.name,
```

Comment:
Any special reason why you are using this construct? It's correct but i feel its unnecessary. And if we are to use this construct, then we should use it everywhere.